Fixed issue where large text on small space is truncated

madhavajay opened the pull request:
Font size search goes a few steps deeper and .Fit logic makes more sense
Binary search now goes exactly midway between each branch until termination
Tests updated to use specific optimalFontSize variable

I am not sure what you will think of this commit but the previous code was failing for me on some edge cases and this seems to solve them.

madhavajay committed:
Fixed issue where large text on small space is truncated

Font size search goes a few steps deeper and .Fit logic makes more sense
Binary search now goes exactly midway between each branch until termination
Tests updated to use specific optimalFontSize variable

tbaranes commented on FittableFontLabel/FittableFontLabel iOS Tests/FittableFontLabelTests.swift:
```diff
@@ -24,6 +24,8 @@ class FittableFontLabelTests: XCTestCase {
     // swiftlint:disable:next line_length
     let longText = "Etiam turpis elit, efficitur nec suscipit vitae, aliquam vitae lorem. Aenean mauris tortor, semper vel facilisis nec, facilisis ut neque. Aliquam erat volutpat. Nam eget varius mauris, ut condimentum quam. Duis ex nisi, iaculis in ex ut, lobortis pretium est. Sed non lorem venenatis, fermentum quam aliquet, ultrices arcu. Cum sociis natoque penatibus et magnis dis parturient m ...
+
+  let optimalFontSize: CGFloat = 2.0703125
```
Can you indent it at the same level as the others?

madhavajay replied:
It has been removed in the latest commit. I have now fixed the indentation of the subsequent commit as well.

tbaranes commented:
Yeah, I understand the issue here. Looks like a good solution to me! Thanks 👍 
If you can take a look at the comment, and add an entry in the CHANGELOG, then we will merge it in master

madhavajay committed:
Better Solution, reverted tests to before last commit

Termination on Binary Search now occurs after state check
If .TooSmall return maxSize, else return minSize

Original Tests pass and edge case is fixed

madhavajay commented:
I just improved it actually, take another a look and let me know if you agree with the updated version, then ill update the CHANGELOG. 😎

madhavajay commented:
I'm a bit confused by the versioning, the CHANGELOG only has Next, but the podspec in main already has 1.3.0. Ill leave that to you and just add the bug fix from these commits.

madhavajay committed:
Updated CHANGELOG

madhavajay committed:
Fixed indentation and added a comment

tbaranes commented:
Ups, my bad, I forgot to update the CHANGELOG after the last released 😬 
I will at  take a look at your change tonight 👍

tbaranes commented:
Just merged it, and fix the CHANGELOG. Let me know you need your fix on master too, otherwise, it will be released at the same time as Swift3.

madhavajay commented:
Awesome thanks. No rush im using my fork for now. 😎
